src/messaging/Message.ts

```diff
@@ -77,35 +94,40 @@ export enum MessageBodyType {
 }
 
 export default class Message implements IMessage {
-  public static ensureOwnerIsSender(message: IMessage): void {
-    switch (message.body.type) {
+  /**
+   * Verifies that the sender of a [[Message]] is also the owner of it, e.g. the owner's and sender's public keys match.
+   *
+   * @param message The [[Message]] object which needs to be decrypted.
+   * @param message.body The body of the [[Message]] which depends on the [[MessageBodyType]].
+   * @param message.senderAddress The sender's public SS58 address of the [[Message]].
+   *
+   */
+  public static ensureOwnerIsSender({ body, senderAddress }: IMessage): void {
```

Comment:
It was not obvious what was actually needed from the message. This way, you still input a message but when looking at the function, you immediately know the required fields from message.